tests/Gelf/Test/Transport/StreamSocketClientTcpTest.php

```diff
@@ -122,35 +129,45 @@ public function testRead(): void
         self::assertEquals(strlen($testData), $numBytes);
 
         // lower timeout for server-socket
-        stream_set_timeout($this->serverSocket, 0, 100);
+        self::assertTrue(stream_set_timeout($this->serverSocket, 0, 100));
 
         $connection = stream_socket_accept($this->serverSocket);
 
         // return input as output
-        stream_copy_to_stream($connection, $connection, strlen($testData));
+        self::assertEquals(strlen($testData), stream_copy_to_stream($connection, $connection, strlen($testData)));
+
+        usleep(300);
+
+        self::assertTrue(fclose($connection));
+
+        usleep(500);
 
-        fclose($connection);
         $readData = $this->socketClient->read($numBytes);
 
         self::assertEquals($testData, $readData);
     }
 
     public function testReadContents(): void
     {
-        $testData = str_repeat("0123456789", mt_rand(1, 10));
+        $testData = str_repeat("0123456789", random_int(1, 10));
+        #$testData = str_repeat("0123456789", 10);
 
         $numBytes = $this->socketClient->write($testData);
         self::assertEquals(strlen($testData), $numBytes);
 
         // lower timeout for server-socket
-        stream_set_timeout($this->serverSocket, 0, 100);
+        self::assertTrue(stream_set_timeout($this->serverSocket, 0, 100));
 
         $connection = stream_socket_accept($this->serverSocket);
 
         // return input as output
-        stream_copy_to_stream($connection, $connection, strlen($testData));
+        self::assertEquals(strlen($testData), stream_copy_to_stream($connection, $connection, strlen($testData)));
+
+        usleep(300);
```

Comment:
These timeouts repeat across multiple tests, it may be worth tp put them behind a descriptive constant? At least add a comment why that "random" sleep is required and how you chose the length of the timeout.

Reply:
I hope to remove this, after root cause is found. Because that depends froms system to system i guess. I was just playing around with the numbers to test it.